Tests/RingBufferTestsKiwi.m

```diff
@@ -0,0 +1,129 @@
+#import <Kiwi/Kiwi.h>
+#import "_RPTRingBuffer.h"
+#import "_RPTMeasurement.h"
+
+@interface _RPTRingBuffer ()
+@property (nonatomic) NSArray<_RPTMeasurement *> *measurements;
+@end
+
+SPEC_BEGIN(RingBufferTestsKiwi)
+
+describe(@"RingBuffer", ^{
+
+    describe(@"init", ^{
+
+        it(@"should be non-nil when initialized with non-zero size", ^{
+            _RPTRingBuffer *buffer = [[_RPTRingBuffer alloc] initWithSize:12];
+            [[buffer should] beNonNil];
+        });
+
+        it(@"should have same size with size passed in parameter", ^{
+            _RPTRingBuffer *buffer = [[_RPTRingBuffer alloc] initWithSize:12];
+            [[buffer.measurements should] haveLengthOf:12];
+        });
+
+        it(@"should set the size with the value passed in parameter", ^{
+            _RPTRingBuffer *buffer = [[_RPTRingBuffer alloc] initWithSize:12];
+            [[theValue(buffer.size) should] equal:theValue(12)];
+        });
+
+        it(@"should be nil when initialized with zero size", ^{
+            _RPTRingBuffer *buffer = [[_RPTRingBuffer alloc] initWithSize:0];
+            [[buffer should] beNil];
+        });
+
+    });
+
+    describe(@"retrieve measurement", ^{
+
+        __block _RPTRingBuffer *buffer = nil;
+        NSUInteger size = 10;
+
+        beforeAll(^{
+            NSMutableArray *builder = [NSMutableArray arrayWithCapacity:size];
+            for (NSUInteger index = 0; index < size; ++index)
+            {
+                _RPTMeasurement *measurement = [_RPTMeasurement new];
+                measurement.trackingIdentifier = index;
+                [builder addObject:measurement];
+            }
+
+            buffer = [[_RPTRingBuffer alloc] initWithSize:size];
+            buffer.measurements = builder.copy;
+        });
+
+        context(@"measurementAtIndex", ^{
+
+            it(@"should return a non-nil measurement when the passed index is less than the buffer's size", ^{
+                NSUInteger validIndex = size-1;
+                _RPTMeasurement *measurement = [buffer measurementAtIndex:validIndex];
+                [[measurement should] beNonNil];
+            });
+
+            it(@"should return a measurement having the same tracking identifier as expected", ^{
+                NSUInteger validIndex = size-1;
+                _RPTMeasurement *measurement = [buffer measurementAtIndex:validIndex];
+                [[theValue(measurement.trackingIdentifier) should] equal:theValue(validIndex)];
```

Comment:
Getting a measurement at validIndex and then comparing trackingIdentifier to validIndex seems confusing.

Also it's not clear why trackingIdentifier should equal validIndex.